index.js

```diff
@@ -1,74 +1,278 @@
-var analyze = require('./lib/analyze-raptor');
-var Promise = require('promise');
-var fs = require('fs');
-var debug = require('debug')('raptor-regression');
-
-var metric = 'visuallyLoaded';
-
-var readFile = function(dataFile) {
-  debug('Processing %s', dataFile);
-  return new Promise(function(resolve, reject) {
-    fs.readFile(dataFile, { encoding: 'utf8' }, function(err, content) {
-      if (err) {
-        return reject(err);
-      }
-
-      var data = JSON.parse(content.split('\n'));
-      resolve(data);
-    });
-  });
+var R = require('ramda');
+
+var BACKWINDOW_SIZE = 12;
+var FOREWINDOW_SIZE = 12;
+var T_THRESHOLD = 7;
+
+/**
+ * Always get 1
+ * @returns 1
+ */
+var single = R.always(1);
+
+/**
+ * Always get zero
+ * @returns 0
+ */
+var zero = R.always(0);
+
+/**
+ * Always get Infinity
+ * @returns Infinity
+ */
+var infinity = R.always(Infinity);
+
+/**
+ * Get the square of a number
+ * @param {Number}
+ * @returns {Number}
+ */
+var square = R.partialRight(Math.pow, 2);
+
+/**
+ * Square every number in a collection
+ * @param {Array<Number>} numbers
+ * @returns {Array<Number>}
+ */
+var squareAll = R.map(square);
+
+/**
+ * Get the numerator for a variance calculation by summing the squares of a collection
+ * @param {Array<Number>} numbers
+ * @returns {Number}
+ */
+var varianceNumerator = R.pipe(squareAll, R.sum);
+
+/**
+ * Get the denominator for a variance calculation by decrementing the length of a collection
+ * @param {Array} array
+ * @returns {Number}
+ */
+var varianceDivisor = R.pipe(R.length, R.dec);
+
+/**
+ * Map over an array with additional arguments for index and original array
+ * @param {Array} array
+ * @returns {Array}
+ */
+var mapIndexed = R.addIndex(R.map);
+
+/**
+ * For given collections of values and weights, multiply each pair and sum together all the products
+ * @param {Array<Number>} values
+ * @param {Array<Number>} weights
+ * @returns Number
+ */
+var calculateWeightedSum = R.pipe(R.zip, R.map(R.partial(R.apply, R.multiply)), R.sum);
```

Comment:
Alright so R.apply takes two args, the set of values and the operation to apply to them. I know one argument is from R.map, which is each element from R.zip, which is a new array of value, weight. Where does the other argument for R.apply come from? Or is that from R.partial... /me confused